src/diagram/components/ui/expandable-input.tsx

```diff
@@ -84,16 +93,17 @@ export const ExpandableInput = ({
     }
   };
 
+  useEffect(() => {
+    if (disabled && value) {
+      setHasLongValue(!!(value.toString().length >= lengthToExpand));
```

Comment:
Using a function seems like a solid suggestion, thanks, but for some reason I can't figure out, it doesn't work. If I try to check the value via a function, the fields won't expand when the value gets too long. I tried several variations on the function idea, but had no luck.

I believe `hasLongValue` has to be a state variable because otherwise it couldn't be set by the `useEffect` that's needed to update disabled fields. I'm not sure if that `useEffect` is the best option for making sure disabled fields are updated, but it's the only way I was able to make it work.

Reply:
I'd have to take a look at what you tried to know why the function didn't work. I think it should be possible to make it work by adding `const checkLongValue = (value: number | undefined) => value?.toString().length >= lengthToExpand;` near the top of the functional component (above the `useState`s), then use that function when you were directly checking the length (lines 29, 37, and 98).

I guess I'm not sure what the `useEffect` is doing. From my understanding, it will change `hasLongValue` when `disabled` or `value` changes (I don't think `lengthToExpand` should ever change). But those are props, so the component should re-render when either of them changes anyway. So it seems to me like you should be able to replace the `useState` that defines `hasLongValue` with a constant: `const hasLongValue = value?.toString().length >= lengthToExpand;`. However, it's definitely possible that I'm misunderstanding something.

(I thought that `!!` is necessary there, but it looks like it isn't because `undefined >= lengthToExpand` evaluates to false, not undefined. Good to know!)